server/app.py

```diff
@@ -12,4 +12,77 @@
 app.register_blueprint(ping_handler)
 
 db = SQLAlchemy(app)
-migrate = Migrate(app, db)
\ No newline at end of file
+migrate = Migrate(app, db)
+
+from models import List, Product
+
+@app.route('/create_list', methods=['POST'])
+def createList():
+    if request.method == 'POST':
+        body = json.loads(request.get_data())
+        list_item = List(body['product_type'])
+        db.create_all()
+        db.session.add(list_item)
+        try:
+            db.session.commit()
+        except:
+            return jsonify({'response': "{} already exists in the database".format(body['product_type'])}), 400
+    return jsonify({'response': "{} was successfully added to the database".format(body['product_type'])}), 200
+
+
+@app.route('/add_product', methods=['GET', 'POST'])
+def addProduct():
+    if request.method == "GET":
+        product_types = List.query.all()
+        ls = []
+        for item in product_types:
+            ls.append(
+                {
+                    "id": item.id,
+                    "product_type": item.product_type
+                })
+        return jsonify({'list': ls}), 200
+
+    elif request.method == 'POST':
+        body = json.loads(request.get_data())
+        product_item = Product(
+            body['list_id'], body['name'], body['description'], body['price'], body['url'])
+        db.create_all()
+        db.session.add(product_item)
+        try:
+            db.session.commit()
+        except:
+            return jsonify({'response': "{} already exists in the database".format(body['name'])}), 400
+    return jsonify({'response': "{} was successfully added to the database".format(body['name'])}), 200
+
+
+@app.route('/edit_list', methods=['GET', 'DELETE'])
+def editList():
+
+    if request.method == "GET":
+        body = json.loads(request.get_data())
+        list_id = body['list_id']
+        products = Product.query.filter_by(list_id=list_id)
+        product_type = List.query.filter_by(id=list_id).first()
+        ls = []
+        for prod in products:
+            ls.append(
+                {
+                    "id": prod.id,
+                    "name": prod.name,
+                    "description": prod.description,
+                    "price": prod.price,
+                    "url": prod.url
+                })
+        return jsonify({product_type.product_type: ls}), 200
+    
+    if request.method == "DELETE":
+        body = json.loads(request.get_data())
+        product_id = body['product_id']
+        product_name = Product.query.filter_by(id=product_id).first().name
+        Product.query.filter_by(id=product_id).delete()
+        db.session.commit()
+        return jsonify({'response': "{} was successfully deleted from the database".format(product_name)}), 200
+
+
```

Comment:
I tried to create the routes based on the 3 provided views given in the [google drive](https://drive.google.com/drive/folders/1_-AKG8TNKAqSrK9E7cs_Z5gLcQYtnMny) because some views call for different types of request (eg. create_list you need to get a list of all product_types)

Reply:
so I was thinking there was 3 different routes each representing a view

Reply:
@aeciorc How do you think I should best set-up the routing. I set them up by the different views of the dashboard, but I assume it wouldn't matter because the dialog views don't actually change the webpages